packages/near-membrane-dom/src/browser-realm.ts

```diff
@@ -167,4 +179,109 @@ function createIframeVirtualEnvironment(
+        }
+        defaultGlobalOwnKeys = getFilteredGlobalOwnKeys(sourceShapeOrOneTimeWindow);
+        if (sourceIsIframe) {
+            ReflectApply(ElementProtoRemove, sourceShapeOrOneTimeWindow, []);
+        }
+        defaultGlobalPropertyDescriptorMap = {
+            __proto__: null,
+        } as unknown as PropertyDescriptorMap;
+        assignFilteredGlobalDescriptorsFromPropertyDescriptorMap(
+            defaultGlobalPropertyDescriptorMap,
+            ObjectGetOwnPropertyDescriptors(globalObjectVirtualizationTarget)
+        );
+        for (let i = 0, { length } = defaultGlobalOwnKeys; i < length; i += 1) {
+            defaultGlobalOwnKeysRegistry[defaultGlobalOwnKeys[i]] = true;
+        }
+        for (const key in defaultGlobalPropertyDescriptorMap) {
+            if (!(key in defaultGlobalOwnKeysRegistry)) {
+                delete defaultGlobalPropertyDescriptorMap[key];
+            }
+        }
+    }
+    const blueRefs = getCachedGlobalObjectReferences(globalObjectVirtualizationTarget);
+    // Create a new environment.
+    const env = new VirtualEnvironment({
+        blueConnector,
+        distortionCallback,
+        redConnector: createConnector(
+            ReflectApply(FunctionProtoBind, ShadowRealmProtoEvaluate, [new ShadowRealmCtor()])
+        ),
+        instrumentation,
+    });
+    linkIntrinsics(env, globalObjectVirtualizationTarget);
+    // window
+    env.link('globalThis');
+    // Set globalThis.__proto__ in the sandbox to a proxy of
+    // globalObjectVirtualizationTarget.__proto__ and with this, the entire
+    // structure around window proto chain should be covered.
+    env.remapProto(globalObjectVirtualizationTarget, blueRefs.WindowProto);
+    let unsafeBlueDescMap: PropertyDescriptorMap = defaultGlobalPropertyDescriptorMap;
+    if (globalObjectVirtualizationTarget !== window) {
+        unsafeBlueDescMap = { __proto__: null } as unknown as PropertyDescriptorMap;
+        assignFilteredGlobalDescriptorsFromPropertyDescriptorMap(
+            unsafeBlueDescMap,
+            ObjectGetOwnPropertyDescriptors(globalObjectVirtualizationTarget)
+        );
+        for (const key in unsafeBlueDescMap) {
+            if (!(key in defaultGlobalOwnKeysRegistry)) {
+                delete unsafeBlueDescMap[key];
+            }
+        }
+    }
+    env.remapProperties(blueRefs.window, unsafeBlueDescMap);
+    if (endowments) {
+        const filteredEndowments: PropertyDescriptorMap = {};
+        assignFilteredGlobalDescriptorsFromPropertyDescriptorMap(filteredEndowments, endowments);
+        removeWindowDescriptors(filteredEndowments);
+        env.remapProperties(blueRefs.window, filteredEndowments);
+    }
+    // We remap `blueRefs.WindowPropertiesProto` to an empty object because it
+    // is "magical" in that it provides access to elements by id.
+    env.remapProto(blueRefs.WindowProto, ObjectCreate(blueRefs.EventTargetProto));
+    return env;
+}
+
+export default SUPPORTS_SHADOW_REALM
```

Comment:
@jdalton @rwaldron, I think the change I suggested is still needed. Basically, it is to force using `createIframeVirtualEnvironment` under certain circumstances. Of course, that could be done at the upper layer in locker package, choosing one or the other.

Reply:
I suspect maybe issues, or rollout strategies if needed. e.g.: rolling out ShadowRealm only for internal (first party) sandboxes first.

Reply:
> I suspect maybe issues, or rollout strategies if needed. e.g.: rolling out ShadowRealm only for internal (first party) sandboxes first.

Ah okay.